pennylane/ops/op_math/controlled_decompositions.py

```diff
@@ -91,12 +91,16 @@ def decomp_circuit(op):
             )[0]
         phi, theta, omega = zyz_decomp.single_qubit_rot_angles()
 
-    return [
-        qml.RZ(phi, wires=target_wire),
-        qml.RY(theta / 2, wires=target_wire),
-        qml.MultiControlledX(wires=control_wires + target_wire),
-        qml.RY(-theta / 2, wires=target_wire),
-        qml.RZ(-(phi + omega) / 2, wires=target_wire),
-        qml.MultiControlledX(wires=control_wires + target_wire),
-        qml.RZ((omega - phi) / 2, wires=target_wire),
-    ]
+    angles = [phi, theta / 2, -theta / 2, -(phi + omega) / 2, (omega - phi) / 2]
+    ops = [qml.RZ, qml.RY, qml.RY, qml.RZ, qml.RZ]
+
+    decomp = []
+    for i, (op, angle) in enumerate(zip(ops, angles)):
+        if not qml.math.isclose(angle, 0.0, atol=1e-8, rtol=0):
+            # Add rotation to decomposition if non-trivial
+            decomp.append(op(angle, wires=target_wire))
+        if i % 2 == 1:
+            # Add MultiControlledX to decomposition
+            decomp.append(qml.MultiControlledX(wires=control_wires + target_wire))
```

Comment:
There's a rather fuzzy judgment line between "this is a pattern, let's abstract it!" and "let's just write out the same thing several times in a row."

Due to readability, I might prefer just writing out the same thing five times.

Abstractions like this can be really good when it lowers the number of things (variable names, helper functions, different layers of abstraction) a developer has to be holding in their active memory or more clearly communicates some underlying structure to the solution. We don't necessarily want to *add* to the number of things developers are having to hold in their memory.

Reply:
I updated the code :)